src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -151,16 +152,35 @@ public int CurrentWorkspaceIndex
         {
             get
             {
+                // It is safe to assume that DynamoModel.CurrentWorkspace is
+                // update-to-date.
                 var viewModel = workspaces.FirstOrDefault(vm => vm.Model == model.CurrentWorkspace);
                 var index = workspaces.IndexOf(viewModel);
+
+                // As the getter could aslo be triggered by the change of model,
+                // we need to update currentWorkspaceViewModel here. 
+                if (currentWorkspaceViewModel != viewModel)
+                    currentWorkspaceViewModel = viewModel;
+
                 return index;
             }
             set
```

Comment:
I'm not 100% certain that `currentWorkspaceViewModel` is needed. From a design perspective, a `DynamoViewModel` cannot live without `DynamoModel`, therefore `DynamoViewModel.CurrentWorkspaceIndex` property should always be inferred by `DynamoModel.CurrentWorkspace` and be made a read-only property. The only notion of "current workspace" should live in `DynamoModel` and nowhere else. In another word, if one were to switch the current workspace, it has to go through `SwitchTabCommand` (that needs a better name, I know), there is no second way around it. Is this possible? I'd prefer to examine all the callers that set `DynamoViewModel.CurrentWorkspaceIndex`, and see if they can be updated to use `SwitchTabCommand` instead.